src/ray/core_worker/actor_manager.cc

```diff
@@ -313,7 +313,7 @@ void ActorManager::SubscribeActorState(const ActorID &actor_id) {
 }
 
 void ActorManager::MarkActorKilledOrOutOfScope(
-    std::shared_ptr<ActorHandle> actor_handle) {
+    const std::shared_ptr<ActorHandle> &actor_handle) {
```

Comment:
If the ownership is not transferred, what about `const ActorHandle&`?
Reasons:
- Const reference ensures immutability and non-nullability;
- From my limited programming experience, I almost never see people using `const std::shared_ptr<>`.

Reply:
agree I think const ActorHandle& makes sense, and shared ptr copy avoidance makes basically no effect
But it was just a clang-tidy fix where it's copied and not moved so can be ref, and linters don't consider it to be trivially copyable, i assume because copy means you're still accessing ref counter and doing the atomic increment which can be a little bit of a hit on heavily threaded code (which we don't really have), but still just for the linter.

Later I think it could also make things easier to review to cut down on # of shared ptrs if we see paths where there's no real copies being made, cutting down on # of shared ptrs will make a diff

Reply:
> But it was just a clang-tidy fix where it's copied and not moved so can be ref

I'm sure using ref instead of shared pointer is better, and clang tidy will not complain about it.
My take on clang-tidy is, it's useful to provide good-enough feedbacks, but not perfect; that's why we need `NOLINT`.

Reply:
I'll make a follow-up pr off of this to change all const shared_ptr& that were changed on this to const &, but want to minimize more manual changes on this

Reply:
Feel free to leave a TODO if you plan to followup (just in case we forget :) )